### PR TITLE
Really delete things in fs::RecursiveDelete

### PR DESCRIPTION
**Pull request by numberZero**
Fix bug introduced in 5b31d32.

### CODE REVIEW

File: src/filesys.cpp [hunk 1]
```diff
@@ -154,8 +153,7 @@ bool RecursiveDelete(const std::string &path)
 			}
 		}
 		infostream<<"RecursiveDelete: Deleting directory "<<path<<std::endl;
-		//bool did = RemoveDirectory(path.c_str();
-		bool did = true;
+		bool did = RemoveDirectory(path.c_str();
```
Comment (srifqi) [inline src/filesys.cpp]: Missing brace

**Comment by srifqi** [comment]
Is there any reason why use these function?

**Comment by srifqi** [comment]
Is there any reason why use these functions?

**Comment by nerzhul** [comment]
@numberZero did you bisect to find this commits ? :o :yellow_heart:

File: src/filesys.cpp (hunk 1 above)
Comment (numberZero) [reply src/filesys.cpp]: thanks

**Comment by numberZero** [comment]
@srifqi Are there any portable alternatives? (The functions are necessary for ContentDB; see also #7370).
@nerzhul No, I scanned through the call graph while working on #7370 and found these commented out lines. Then `git blame` told me who and when did that; that was... unexpected, at the very least)

**Comment by numberZero** [comment]
BTW, I couldn’t bisect, as my win32 build system lacks some important parts currently (thanks Arch moving MinGW to [source-code-only] AUR. I tried to compile it, but that’s worse than compiling Linux...)

**Comment by red-001** [comment]
Could you clean up the other code in the function? Just some code style issues and deduplication `IsDir`.

**Comment by sfan5** [comment]
> my win32 build system lacks some important parts currently

https://github.com/minetest/minetest/blob/master/util/travis/before_install.sh#L30

> (thanks Arch moving MinGW to [source-code-only] AUR. I tried to compile it, but that’s worse than compiling Linux...)

it's actually not <i>that</i> hard, you just need to do it in the right order

**Comment by numberZero** [comment]
> Could you clean up the other code in the function? Just some code style issues and deduplicating IsDir.

Done. Haven’t tested, though, as...

> it's actually not that hard, you just need to do it in the right order

...and a few spare GiBs which I haven’t. Linux required just about a GiB, and took only half a day to compile (on 1×1.8 GHz Athlon which I had)

**Comment by paramat** [comment]
Needs someone to test this on Windows.

**Comment by Fixer-007** [comment]
What it fixes exactly? What to check?

**Comment by SmallJoker** [comment]
@Fixer-007 It should fix the world- and probably also the content deletion.

**Comment by Fixer-007** [comment]
Works for me on win7, deletes worlds and mods, thank you!

**Comment by Ezhh** [comment]
Big thank you for this fix. I can finally run current dev without going crazy. :)

**Pull request merged**